pkg/traceql/ast.go

```diff
@@ -751,14 +766,20 @@ func (s Static) Equals(o *Static) bool {
 		return s.Type == o.Type && bytes.Equal(s.valBytes, o.valBytes)
 	case TypeStringArray:
 		return s.Type == o.Type && slices.Equal(s.valStrings, o.valStrings)
-	case TypeNil:
-		return o.Type == TypeNil
 	default:
 		// should not be reached
 		return false
 	}
 }
 
+func (s Static) NotEquals(o *Static) bool {
+	if s.Type == TypeNil || o.Type == TypeNil {
```

Comment:
let's add a test case for this as well because we are handling it in a special way.

Reply:
extended `TestStatic_Equals` to handle